lib/net/dhcp/record.rb

```diff
@@ -10,8 +10,16 @@ def initialize(opts = { })
       self.ip      = Net::Validations.validate_ip! self.ip
     end
 
+    def lease?
+      type && type == "lease"
+    end
+
+    def reservation?
+      !lease?
```

Comment:
Is the record type always available, even in the deprecated proxy API? The reason I'm asking is because all leases will look like reservations as `!lease?` evaluates to true if `type == nil`

Reply:
Implemented second option, so previous implementation stays the same. There is a "legacy" flag and that is used in the codebase.